crates/bevy_ecs/src/query/state.rs

```diff
@@ -467,14 +467,12 @@ where
     /// Runs `func` on each query result for the given [`World`]. This is faster than the equivalent
     /// iter() method, but cannot be chained like a normal [`Iterator`].
     ///
-    /// This can only be called for read-only queries.
```

Comment:
This comment was blatantly false, and seems to just be a mistake. It contradicts the safety comment, and doesn't seem to make sense with the API.